T4MVCHostMvcApp/Controllers/TestAttributeControllers.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Web.Mvc;
+
+namespace T4MVCHostMvcApp.Controllers {
+
+
+
+  //
+  // these are tests for the [T4MVC] attribute
+  // look for the results under the T4MVC.tt node
+  // those with "YES" should appear, those with "NO" should not
+  //
+
+
+
+  /// test regular controllers' names ...................................................................
+
+  // this controller should be processed, and its "Controller" substring removed
+  public partial class TestAttribute_YES_Name1Controller : Controller { }
```

Comment:
Getting a lot of Code Analysis violations in this file. e.g. this line causes:

    CA1707	Identifiers should not contain underscores	Remove the underscores from type name 'TestAttribute_YES_Name1Controller'.

Search for `SuppressMessage` in the sources to see how this is handled.